src/main/java/com/mitchej123/hodgepodge/mixins/late/ic2/MixinTEReactorCacheReactorSize.java

```diff
@@ -0,0 +1,69 @@
+package com.mitchej123.hodgepodge.mixins.late.ic2;
+
+import net.minecraft.tileentity.TileEntity;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import ic2.api.Direction;
+import ic2.core.ContainerBase;
+import ic2.core.block.reactor.tileentity.TileEntityNuclearReactorElectric;
+import ic2.core.block.reactor.tileentity.TileEntityReactorChamberElectric;
+
+@Mixin(value = TileEntityNuclearReactorElectric.class)
+public class MixinTEReactorCacheReactorSize extends TileEntity {
+
+    @Unique
+    private short hodgepodge$reactorSize = -1;
+
+    @Inject(
+            method = "updateEntityServer",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lic2/core/block/reactor/tileentity/TileEntityNuclearReactorElectric;getReactorSize()S",
+                    shift = At.Shift.BEFORE,
+                    ordinal = 0),
+            remap = false)
+    private void hodgepodge$resetReactorSize(CallbackInfo ci) {
+        hodgepodge$reactorSize = -1;
+    }
+
+    @Inject(method = "getGuiContainer", at = @At("HEAD"), remap = false)
+    private void hodgepodge$resetReactorSizeOnGui(
+            CallbackInfoReturnable<ContainerBase<TileEntityNuclearReactorElectric>> cir) {
+        // Reset reactor size when opening the GUI, so that it is recalculated immediately
+        hodgepodge$reactorSize = -1;
+    }
+
+    /**
+     * @author kuba6000
+     * @reason cache reactor size to reduce lags
+     */
+    @Overwrite(remap = false)
```

Comment:
just so you know to avoid overwriting the whole method, you can inject at HEAD and then use a ModifyReturnValue to capture the returned value. It's best practice to avoid overwriting a whole method when possible